### PR TITLE
Dashboard bugs

### PR DESCRIPTION
**Pull request by milansar**
BW - 137,138,140

### CODE REVIEW

File: src/app/components/side-nav/side-nav.component.ts [hunk 1]
```diff
@@ -17,11 +18,14 @@ export class SideNavComponent implements OnDestroy {
...
-  constructor(private store: Store<AppState>) {
+  constructor(private store: Store<AppState>, private router: Router) {
     this.userDict$ = store.select(appState.coreState).select(s => s.userDict);
     this.subs.push(this.userDict$.subscribe(userDict => this.userDict = userDict));
   }
   ngOnDestroy() {
     Utils.unsubscribe(this.subs);
   }
+  navigateUrl() {
+    this.router.navigate(['my/questions', this.user.userId]);
```
Comment (anihalaney) [inline src/app/components/side-nav/side-nav.component.ts]: why do we need userId in the route?

File: src/app/user/components/user-stats-card/user-stats-card.component.html [hunk 2]
```diff
@@ -4,53 +4,53 @@
   </ul>
   <ul id="stats" class="tab-content active">
     <li>
-      <a href="#">
+      <!-- <a href="#"> -->
```
Comment (anihalaney) [inline src/app/user/components/user-stats-card/user-stats-card.component.html]: remove commented code

File: src/app/components/side-nav/side-nav.component.ts (hunk 1 above)
Comment (saloniss1345) [reply src/app/components/side-nav/side-nav.component.ts]: We are loading user's questions all router redux based on userId.So, we have added userId in route.

Comment (anihalaney) [reply src/app/components/side-nav/side-nav.component.ts]: userid is available in the store. we don't need it in the route. please check with @milansar or @yugank1991 on how to implement this.

Comment (milansar) [reply src/app/components/side-nav/side-nav.component.ts]: I feel current implementation is correct, we are loading user questions using NGRX Router navigation in effects though store has userID but not sure how to access store from effects

please check following code in src\app\user\store\effects\user.effects.ts
 // handle location update
    loadUserPublishedRouteQuestions$ = this.actions$
        .ofType('ROUTER_NAVIGATION')
        .map((action: any): RouterStateUrl => action.payload.routerState)
        .filter((routerState: RouterStateUrl) =>
            routerState.url.toLowerCase().startsWith('/my/questions') &&
            routerState.params.userid
        )
        .pipe(
        switchMap((routerState: RouterStateUrl) =>
            this.questionService.getUserQuestions(routerState.params.userid, true).pipe(
                map((questions: Question[]) =>
                    new userActions.LoadUserPublishedQuestionsSuccess(questions)
                )
            )
        )
        );

Comment (anihalaney) [reply src/app/components/side-nav/side-nav.component.ts]: I'll have to look at the code, but you can get inject the store into the effect and it'll be something like:
let userid: string;
this.store.take(1).subscribe(s => userid = s.user);

Note that if you have it in the url, then I can copy that url and paste it when I'm logged in as someone else and it'll break things.

File: src/app/user/store/effects/user.effects.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ import { coreState } from '../../../core/store';
 @Injectable()
 export class UserEffects {
 
+    user: User;
```
Comment (anihalaney) [inline src/app/user/store/effects/user.effects.ts]: we don't need class level variable

File: src/app/user/store/effects/user.effects.ts [hunk 4]
```diff
@@ -30,32 +31,44 @@ export class UserEffects {
...
+    loadUserPublishedRouteQuestions$ = this.actions$
+        .ofType('ROUTER_NAVIGATION')
+        .map((action: any): RouterStateUrl => action.payload.routerState)
+        .filter((routerState: RouterStateUrl) =>
+            routerState.url.toLowerCase().startsWith('/my/questions')
+        ).pipe(
+        switchMap((routerState: RouterStateUrl) => {
+            this.store.select(coreState).select(s => s.user).take(1).subscribe(user => {
+                if (user) {
+                    this.user = user;
```
Comment (anihalaney) [inline src/app/user/store/effects/user.effects.ts]: use local variable
if the "if" condition is not true won't it give an error? why do we even have the if condition?

**Pull request merged**